spec/models/tenant_quota_spec.rb

```diff
@@ -1,5 +1,116 @@
 require 'spec_helper'
 
 describe TenantQuota do
-  pending "add some examples to (or delete) #{__FILE__}"
+  let(:settings) {{}}
+  let(:tenant)   {Tenant.new(:domain => 'x.com', :parent => default_tenant)}
+
+  let(:default_tenant) do
+    Tenant.seed
+    Tenant.default_tenant
+  end
+
+  let(:root_tenant) do
+    Tenant.seed
+    Tenant.root_tenant
+  end
+
+  context "validations" do
+    it "rejects invalid name" do
+      expect { described_class.create!(:name => "XXX") }.to raise_error
+    end
+
+    it "rejects missing value" do
+      expect { described_class.create!(:name => :cpu_allocated, :unit => "mhz") }.to raise_error
+    end
+
+    it "rejects missing unit" do
+      expect { described_class.create!(:name => :cpu_allocated, :value => 4096) }.to raise_error
+    end
+  end
+
+  describe ".set" do
+    it "adds new quotas" do
+      described_class.set(default_tenant, :cpu_allocated => {:value => 1024}, :vms_allocated => {:value => 20}, :mem_allocated => {:value => 4096})
```

Comment:
I don't follow this.  Why not use activerecord to build these instead of passing hashes?

I would expect to set quotas via `tenant.tenant_quotas.create(:name => :cpu_allocated, :value => 1024)`

It feels like the UI will have several places where they pass around hashes that know our internal structure.